pkg/proxy/ipvs/proxier.go

```diff
@@ -2079,15 +2086,25 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 	}
 
 	for _, epInfo := range endpoints {
-		if !epInfo.IsReady() {
-			continue
+		if epInfo.IsReady() {
+			readyEndpoints.Insert(epInfo.String())
 		}
 
-		if onlyNodeLocalEndpoints && !epInfo.GetIsLocal() {
-			continue
+		if onlyNodeLocalEndpoints && epInfo.GetIsLocal() {
+			if epInfo.IsReady() {
+				localReadyEndpoints.Insert(epInfo.String())
+			} else if epInfo.IsServing() && epInfo.IsTerminating() {
+				localReadyTerminatingEndpoints.Insert(epInfo.String())
+			}
 		}
+	}
 
-		newEndpoints.Insert(epInfo.String())
+	newEndpoints := readyEndpoints
+	if onlyNodeLocalEndpoints {
+		newEndpoints = localReadyEndpoints
+		if utilfeature.DefaultFeatureGate.Enabled(features.ProxyTerminatingEndpoints) && localReadyTerminatingEndpoints.Len() > 0 {
```

Comment:
Shouldn't you need to be checking here if localReadyEndpoints == 0 instead of localReadyTerminatingEndpoints.Len() > 0? 


This way:
* If you have Ready Local Endpoints, send to them
* Otherwise populate with the terminating ones (if they exists), otherwise it will blackhole them

Reply:
Great catch, fixed and also added an extra test cases for this